File: TerminBA/TerminBA.Models/Request/SportCenterUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TerminBA.Models.Model;

namespace TerminBA.Models.Request
{
    public class SportCenterUpdateRequest
    {
        [Required]
        [MaxLength(100)]
        public string? Username { get; set; }

        [Required]
        [Phone]
        public string? PhoneNumber { get; set; }

        [Required]
        public int CityId { get; set; }

        [Required]
        public required string Address { get; set; }

        [Required]
        public bool IsEquipmentProvided { get; set; }

        [MaxLength(180)]
        public string? Description { get; set; }

        [MinLength(1, ErrorMessage = "At least one sport must be selected.")]
        public List<int>? SportIds { get; set; }

        [MinLength(1, ErrorMessage = "At least one amenity must be selected.")]
        public List<int>? AmenityIds { get; set; }

        [MinLength(1, ErrorMessage = "Working hours are required.")]
        public List<WorkingHoursInsertRequest>? WorkingHours { get; set; }
    }
}

File: TerminBA/TerminBA.UI/terminba_admin_desktop/lib/model/sport_center_update_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:terminba_admin_desktop/model/working_hours_insert_request.dart';

part 'sport_center_update_request.g.dart';

@JsonSerializable()
class SportCenterUpdateRequest {
	String username;
	String phoneNumber;
	int cityId;
	String address;
	bool isEquipmentProvided;
	String description;
	List<int> sportIds;
	List<int> amenityIds;
  List<WorkingHoursInsertRequest> workingHours;

	SportCenterUpdateRequest(this.username,this.phoneNumber,this.cityId,this.address,this.isEquipmentProvided,this.description,this.sportIds,this.amenityIds,this.workingHours);

	factory SportCenterUpdateRequest.fromJson(Map<String, dynamic> json) => _$SportCenterUpdateRequestFromJson(json);

	Map<String, dynamic> toJson() => _$SportCenterUpdateRequestToJson(this);
}